Llvm16

smuzaffar opened the pull request.

hahnjo committed:
[cling] LookupFile takes arguments as ConstSearchDirIterator

hahnjo committed:
[cling] Lexer does not expose getLangOpts() anymore

Store a reference ourselves. Also Lexer::isIdentifierBodyChar() was
renamed to Lexer::isAsciiIdentifierContinueChar().

hahnjo committed:
[cling] TargetRegistry.h moved to MC/

hahnjo committed:
[cling] CompoundStmt::Create takes FPOptionsOverride

hahnjo committed:
[cling] getQualifiedTemplateName takes a TemplateName

hahnjo committed:
[cling] Update for new FileNotFound callback

hahnjo committed:
[cling] Adapt and heavily simplify ClingJITLinkMemoryManager

Now we can just inherit from InProcessMemoryManager and override the
deallocate() method.

hahnjo committed:
[cling] llvm::Optional::hasValue was renamed to has_value

hahnjo committed:
[cling] InclusionDirective takes an clang::OptionalFileEntryRef

hahnjo committed:
[cling] Update InvocationOptions

hahnjo committed:
[interpreter] Require LLVM 16

hahnjo committed:
[cling] Resource directory based on major version only

hahnjo committed:
[cling] Adapt arguments for updated APIs

Forwarding data (differently) or passing trivial parameters.

hahnjo committed:
[cling] Some changes from JITEvaluatedSymbol to ExecutorAddr

hahnjo committed:
[cling] SubstDefaultTemplateArgumentIfAvailable

... takes an additional vector for canonical arguments.

hahnjo committed:
[cling] MultiLevelTemplateArgumentList

... takes a Decl and a boolean argument Final.

hahnjo committed:
[cling] Misc renamings and type changes

hahnjo committed:
[cling] Misc adaptations for updated APIs

hahnjo committed:
[cling] Update ClingMemoryManager with llvm::Align

hahnjo committed:
[cling] Remove PMBuilder members

These two fields don't exist anymore in LLVM 16.

hahnjo committed:
[cling] Don't print name of EnumDecl if there is none

The function overload without PrintingPolicy is not available in
LLVM 16, see https://reviews.llvm.org/D134813.

hahnjo committed:
[cling] Adapt to updated forgetDecl signature

The GlobalDecl parameter wasn't used since quite some time.

hahnjo committed:
[cling] Update handling of ElaboratedType in GetFullyQualifiedType

In LLVM 16, we now get ElaboratedType far more often, but they don't
have all namespace qualifiers we need. This fixes the build of the
RooFitCore module, but many other places are likely affected as well.

hahnjo committed:
[cling] Remove extension of legacy pass manager

adjustPassManager was only used for a few backends, such as AMDGPU
and NVPTX, but not on the CPU architectures we care about.
The DiscriminatorsPass was also removed upstream Clang, see commit
https://github.com/llvm/llvm-project/commit/2978d02681.

hahnjo committed:
[cling] Update registration of atexit handlers

GenericLLVMIRPlatformSupport now declares atexit, which leads to a
duplicate symbol error during startup and fails the transaction.
Completely switch to the provided infrastructure and just override
the helpers to register the functions with our Interpreter.

hahnjo committed:
[cling] Set MSVolatile in LangOpts

hahnjo committed:
[cling] Handle UsingType in GetPartiallyDesugaredType

According to https://github.com/llvm/llvm-project/commit/af27466c50,
it is used to represent types pulled in with a using declaration, for
example `using std::error_code; error_code x;`.

This fixes the build of the ROOTTMVASofie module on macOS where this
was causing `std::size_t` not be fully desugared to `unsigned long`
at first to later end up with two identical classes in a list that
must not have duplicates.

vgvassilev committed:
[cling] Handle llvm::Errors correctly.

This fixes several tests on osx.

vgvassilev committed:
[cling] Correctly infer the SDKROOT.

This fixes the -lSystem issues we saw on recent osx platforms.

hahnjo committed:
[cling] Remove expected space before array brackets in tests

jblomer committed:
[ntuple] add chain support for RDF data source

hahnjo committed:
[cling] Handle ElaboratedType in ReSubstTemplateArg

In LLVM 16, we get ElaboratedType's in TemplateArgument's and need
to recurse into them. This fixes the handling of Double32_t in the
test roottest-root-io-double32-make.

hahnjo committed:
Update expected template names in TClingMethodInfoTests.cxx

LLVM now fully qualifies template template parameters when printing,
since commit https://github.com/llvm/llvm-project/commit/5fb3f43778
As pointed out in that message, this makes it consistent with type
parameters, so add an additional test (Yd1) that would have already
passed with LLVM 13.

hahnjo committed:
[cling] Remove elaborated keyword from output checks

LLVM 16 honors how the declarations were written, without the class
or struct keywords.

hahnjo committed:
[cling] Update expected error messages in tests

hahnjo committed:
[cling] Update expected types in output checks

hahnjo committed:
[win64] Disable Cling optimizations

With the upgrade to LLVM 16, we again have test failures due to crashes
and output ordering compared to the reference files.

This reverts commit 47277dffd5e6c99f3b09e4e66e524c4f4f8d8f3b.

hahnjo committed:
[cling] Fix preprocessor option parsing

Exclude options::DXCOption to avoid confusing the parser with the
second -E option.

hahnjo committed:
[cling] Expect normalized alias option offload_device_only

See also commit de84bfc508 during the previous LLVM upgrade...

hahnjo committed:
[cling] Expect additional error in Prompt/BlockComments.C

bellenot committed:
[TCling] Prepend drive letter to absolute path

This seems to be required with LLVM 16 on Windows and fixes
roottest/root/meta/callfunc/assertUnload.C

hahnjo committed:
[cling] Avoid unloading Decls not found in DeclContext

This fixes roottest/root/meta/tuple/exec_configtype_unload.C on Windows.

Co-authored-by: Bertrand Bellenot <Bertrand.Bellenot@cern.ch>

hahnjo committed:
[cling] Desugar UnaryTransformType

This fixes PyROOT's handling of smart pointers, such as shared_ptr,
with the new macOS SDK 14.0 resulting in a remove_extent transform.

Commit:
Add RNTuple merger prototype

Commit:
Apply clang-format

Commit:
First set of review comments

Commit:
Rework source to destination column id mapping logic and add new tests for merging sources with mismatched fields

Commit:
Change merge function signature based on review comments

Commit:
Apply clang-format

Commit:
Update collection of columns

Commit:
Add a test for merging vector types

Commit:
Extend matching to include type and version in addition to name

Commit:
Add a new test to check the behavior when merging fields w/ the same name but inconsistent types

lmoneta committed:
[tmva][sofie] Add support for variable number of nodes and edges

enirolf committed:
[ntuple] Use `ROOT::Experimental::EColumnType` namespace in inspector tests

enirolf committed:
[ntuple] Add `RNTupleInspector::GetPageSizeDistribution`

jblomer committed:
[ntuple] improve RDF MT work scheduling

Instead of parallelizing just within any given file of the chain, the
work scheduling will now
  - Use one slot per file as long as enought files are available
  - Otherwise, distribute slots evenly on the remaining files and
    distribute, per file, the slots evenly and aligned on cluster
    boundaries

jblomer committed:
[ntuple] prevent per-range recreation of values

jblomer committed:
[ntuple] fix handling of principal source in RDF source

hahnjo committed:
[ntuple] Introduce RPagePersistentSink

The plan is to move functionality from RPageSink that is only relevant
for "persistent" sinks, ie RPageSinkDaos and RPageSinkFile but not for
RPageSinkBuf.

hahnjo committed:
[ntuple] Throw in RPageSinkBuf::CommitSealedPageImpl

RPageSinkBuf is supposed to buffer pages, compress / seal them, and
then forward them to an inner sink when committing a cluster. This
cannot work with sealed pages, so this method should never be called
on a RPageSinkBuf.

hahnjo committed:
[ntuple] Move Commit implementations to RPagePersistentSink

They are not needed in RPageSinkBuf, that was previously forced to
override the *Impl() methods.

hahnjo committed:
[ntuple] Avoid calling RPageSink::UpdateSchema from RPageSinkBuf

RPageSinkBuf never serializes a descriptor, so it only needs a very
minimal traveral of the fields to call ConnectPageSink.

hahnjo committed:
[ntuple] Move descriptor building to RPagePersistentSink

They are not needed in RPageSinkBuf, for example.

hahnjo committed:
[ntuple] Move some data members into RPagePersistentSink

They are all related to building the descriptors and serializing the
on-disk formats.

hahnjo committed:
[ntuple] Move metric counters into RPagePersistentSink

hahnjo committed:
[ntuple] Use only one RPageStorage::fMetrics

Subclasses can still register their own counters.

lmoneta committed:
[tmva][sofie] Use RTensor<int> for the GNN edge index data

Instead of a vector of receivers and senders use a common tensor of shape (2, number of edges) for the edge index conections (receivers senders).
This is in line with pytorch geomoetric data structure

Improve tutorial for parsing the model and for run the model

Fix also tutorial dependency

enirolf committed:
[NFC][ntuple] Improve `RNTupleImporter` docs formatting

hahnjo committed:
Import ROOT-llvm16-20231130-01

guitargeek committed:
[RF] RooFit code modernization

In particular, delete the `RooTFoamBinding` source files. The class is
only used in `RooFoamGenerator`, so we can just define the
`RooTFoamBinding` class in `RooFoamGenerator.cxx`.

smuzaffar commented:
test parameters:
- full_cmssw = true
- pull_request = cms-externals/boost#6

smuzaffar commented:
please test

cmsbuild commented:
A new Pull Request was created by  @smuzaffar (Malik Shahzad Muzaffar) for branch cms/master/83d9889d7e.

@iarspider, @aandvalenzuela, @cmsbuild, @smuzaffar can you please review it and eventually sign? Thanks.
@sextonkennedy, @rappoccio, @antoniovilela you are the release manager for this.
cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild commented:
-1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0b56f5/36240/summary.html
**COMMIT**: cc55a9d7fd3e0dcd63bb173cba2d8e23620e00eb
**CMSSW**: CMSSW_14_0_X_2023-11-30-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/root/189/36240/install.sh` to create a dev area with all the needed externals and cmssw changes.


## External Build

I found compilation error when building: 

<pre>File "/data/cmsbld/jenkins/workspace/ib-run-pr-tests/pkgtools/scheduler.py", line 267, in doSerial
result = commandSpec[0](*commandSpec[1:])
File "./pkgtools/cmsBuild", line 3701, in installPackage
File "./pkgtools/cmsBuild", line 3452, in installRpm
RpmInstallFailed: Failed to install package root. Reason:
error: Failed dependencies:
	/usr/bin/python is needed by lcg+root+6.26.11-450c552b334c6c6c6ce355bad6de4fb4-1-1.x86_64

* The action "build-cms+dqmgui+1.0.1_R6.26.11-34632a6ac0a967b4c022a3290e940479" was not completed successfully because The following dependencies could not complete:
install-lcg+root+6.26.11-450c552b334c6c6c6ce355bad6de4fb4
* The action "build-cms+cmssw-tool-conf+57.0-053f0623d2ec934a237598e0c1386b39" was not completed successfully because The following dependencies could not complete:
</pre>

smuzaffar commented:
please test for CMSSW_14_0_ROOT6_X

guitargeek committed:
[RF] Apply `clang-tidy` suggestions also to header files

cmsbuild commented:
-1

**Failed Tests**:   Build 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0b56f5/36246/summary.html
**COMMIT**: cc55a9d7fd3e0dcd63bb173cba2d8e23620e00eb
**CMSSW**: CMSSW_14_0_ROOT6_X_2023-11-30-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/root/189/36246/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @fwyzard cms-sw/cmsdist#8848

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0b56f5/36246/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0b56f5/36246/git-merge-result


## Build

I found compilation error when building: 

<pre>>> Building binary testConditionDatabase_4
Copying tmp/el8_amd64_gcc12/src/CondCore/CondDB/test/testConditionDatabase_4/testConditionDatabase_4 to productstore area:
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_14_0_ROOT6_X_2023-11-30-2300/src/CondCore/CondDB/test/testConnectionPool.cpp
>> Building binary testConnectionPool
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/../lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: cannot find -lCondFormatsRunInfo: No such file or directory
collect2: error: ld returned 1 exit status
>> Deleted: tmp/el8_amd64_gcc12/src/CondCore/CondDB/test/testConnectionPool/testConnectionPool
gmake: *** [tmp/el8_amd64_gcc12/src/CondCore/CondDB/test/testConnectionPool/testConnectionPool] Error 1
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_14_0_ROOT6_X_2023-11-30-2300/src/CondCore/CondDB/test/testFrontier.cpp
>> Building binary testFrontier
Copying tmp/el8_amd64_gcc12/src/CondCore/CondDB/test/testFrontier/testFrontier to productstore area:
</pre>

smuzaffar commented:
please test for CMSSW_14_0_ROOT6_X

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0b56f5/36257/summary.html
**COMMIT**: cc55a9d7fd3e0dcd63bb173cba2d8e23620e00eb
**CMSSW**: CMSSW_14_0_ROOT6_X_2023-11-30-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/root/189/36257/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @aandvalenzuela cms-sw/cmssw#43462
 - @aandvalenzuela cms-sw/cmssw#43460
 - @aandvalenzuela cms-sw/cmssw#43461
 - @aandvalenzuela cms-sw/cmssw#43464
 - @mmusich cms-sw/cmssw#43469
 - @fwyzard cms-sw/cmssw#43451
 - @bsunanda cms-sw/cmssw#43447
 - @bsunanda cms-sw/cmssw#43419
 - @iarspider cms-sw/cmssw#43399
 - @fwyzard cms-sw/cmsdist#8848

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0b56f5/36257/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0b56f5/36257/git-merge-result


## Comparison Summary



**Summary**:
 - You potentially [removed 271 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2023-11-30-2300+0b56f5/60108/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 3992 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2023-11-30-2300+0b56f5/60108/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 3370032
 - DQMHistoTests: [Total failures: 61953](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2023-11-30-2300+0b56f5/60108/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3308057
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 214 log files, 167 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2023-11-30-2300+0b56f5/60108/triggerResults): found differences in 1 / 48 workflows

jblomer committed:
[ntuple] add optional limits to file-based readv

guitargeek committed:
[RF] Introduce `RooAbsCollection::addTyped()` method

It's a very common pattern in the RooFit implementation and user code to
check the type when adding to a `RooAbsCollection`, and erroring out if
the type is not correct.

This commit suggests a function in `RooAbsCollection` to do exactly
this, drastically reducing the code duplication in RooFit and experiment
frameworks like CMS combine.

lmoneta committed:
[tmva][sofie] Modify sofie_gnn.py test for changes in the GNN_Data for edge indices

guitargeek committed:
[RF] Make `RooFFTConfPdf` also work if ROOT was built with `fftw3=OFF`

Since ROOT 6.30, we are not building ROOT with the `math/fftw`
subpackage anymore (`fftw3=OFF`). It is the interface between ROOT and
fftw3, but is incompatible with fftws GPL license.

That means that for default ROOT builds, the `RooFFTConvPdf` class for
FFT convolutions is not working anymore, because it uses `math/fftw`.

This commit implements a solution to make the `RooFFTConvPdf` work
again in this situation: the routine that uses fftw is declared
on-the-fly to the interpreter. This will work if the user has an
external install of `fftw3`, which is usually available in all Linux
distributions.

jblomer committed:
[io] query server for readv limits in xrd raw file

jblomer committed:
[io] white space fixes in RRawFileNetXNG

guitargeek committed:
[RF] Promote `ROOFIT_MEMORY_SAFE_INTERFACES` to users in release notes

  * The `ROOFIT_MEMORY_SAFE_INTERFACES` macro is explained in the
    release notes, so that users and framework developers can use it to
    detect memory leaks

  * Since people are now encouraged to use it, the
    `RooFit::Detail::owningPtr` helper function was renamed to
    `RooFit::makeOwningPtr`, because this is now not an implementation
    detail anymore but part of the user interface

  * Remove `takeOwnership` constructor arguments from
    `PiecewiseInterpolation` and `RooAddition`, because there are not
    memory-safe and it's not easily possible to only exclude selected
    parameters behind the `ROOFIT_MEMORY_SAFE_INTERFACES` macro without
    also changing the sources. Like this, we ensure that the memory-safe
    interfaces can be used without re-building ROOT.

krasznaa committed:
Explicitly set Vdt_ROOT when builtin_vdt was used for the build.

This way, a default location would be provided for VDT, which the
user could still override if they wanted to.

jblomer committed:
[ntuple] improve member documentation (NFC)

Co-authored-by: Florine de Geus <git@fwdg.nl>

jblomer committed:
[ntuple] improve member documentation (NFC)

Co-authored-by: Florine de Geus <git@fwdg.nl>

jblomer committed:
[ntuple, NFC] improve code comment of GetColumnReadersImpl()

jblomer committed:
[ntuple] rename RNTupleDS::REntryRange --> REntryRangeDS

hahnjo committed:
[cling] Provide fallback for LLVM_INCLUDE_DIRS

In standalone builds, it could otherwise happen that the variable is
not set during the first CMake invocation and tests fail because they
are unable to locate the LLVM headers.

krasznaa committed:
Require CMP0074 to be NEW for ROOTConfig.cmake.

Axel-Naumann committed:
[cling] test/Driver/CommandHistory.C: `env -u` also works on macOS.

jblomer committed:
[ntuple] add support for n-dim C arrays in EDMs

jblomer committed:
[ntuple] update specification (NFC)

guitargeek committed:
[RF] Again skip overflow bins correctly in HistFactory in 2D and 3D case

In the 6.30 dev cycle, I made a mistake when refactoring the HistFactory
code in 3c68044c257:

https://github.com/root-project/root/commit/3c68044c257e9ee7207e23e7cceb69d7eda596e9#diff-42ea09980f1ed2bacde4381f212c35d78be6ddedccf746953dc99d6455525d8aL1801-L1804

The robust way to skip overflow bins in 2D and 3D was replaced with
something that only worked in 1D.

Thanks to Veronica for reporting this on the ROOT forum!

https://root-forum.cern.ch/t/discrepancy-in-2d-histfactory-fit-results-between-root-versions-6-28-06-vs-nightlies-with-beeston-barlow-method-activated

I validated now that the fit results with the reproducer on the forum
are the same with 6.28 and 6.30.

To be backported to the 6.30 branch.

Commit:
Implement Jakob's comments

Commit:
Apply clang-format

jblomer committed:
[ntuple] extend ndim array unit tests

cmsbuild commented:
Pull request #189 was updated.

smuzaffar commented:
please test

cmsbuild commented:
-1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0b56f5/36346/summary.html
**COMMIT**: 99e17086860d1264d545d6fb02d7480dc411a56f
**CMSSW**: CMSSW_14_0_X_2023-12-05-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/root/189/36346/install.sh` to create a dev area with all the needed externals and cmssw changes.


## External Build

I found compilation error when building: 

<pre>File "/pool/condor/dir_3106/jenkins/workspace/ib-run-pr-tests/pkgtools/scheduler.py", line 267, in doSerial
result = commandSpec[0](*commandSpec[1:])
File "./pkgtools/cmsBuild", line 3701, in installPackage
File "./pkgtools/cmsBuild", line 3452, in installRpm
RpmInstallFailed: Failed to install package root. Reason:
error: Failed dependencies:
	/usr/bin/python is needed by lcg+root+6.26.15-8725468ea0e302bf76c73e6b8e0e59ac-1-1.x86_64

* The action "build-cms+dqmgui+1.0.1_R6.26.15-5e20aaea78c12a6bd1792272975e878c" was not completed successfully because The following dependencies could not complete:
install-lcg+root+6.26.15-8725468ea0e302bf76c73e6b8e0e59ac
* The action "build-cms+cmssw-tool-conf+57.0-65c4d9f1923861ae3f9e5063015dc40a" was not completed successfully because The following dependencies could not complete:
</pre>

smuzaffar commented:
please test for CMSSW_14_0_ROOT6_X

vepadulano committed:
[RF] Enable external roottest build

By moving the header currently hidden in `src/RooFit` to `inc/RooFit` we can
allow a roottest build to work when built with an external installation of ROOT.
To preserve the intent of the header, the functions declared are moved to the
`RooFit::Internal` namespace, as they are not meant for public use.

lmoneta committed:
[tmva][sofie] Fix failing GNN Python tutorial and dependency

Fix the dependency of the GNN C++ Application tutorial from the Python parser tutorial. Add Parser tutorial directly by hand since previous way of secifying the dependency did not work in this case.

Need to specify the cling include pathin TMVA_SOFIE_GNN_Application.C to be sure that the generated headers are found whne running from ctest

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-0b56f5/36348/summary.html
**COMMIT**: 99e17086860d1264d545d6fb02d7480dc411a56f
**CMSSW**: CMSSW_14_0_ROOT6_X_2023-12-05-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/root/189/36348/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [added 100 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2023-12-05-2300+0b56f5/60188/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 3987 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2023-12-05-2300+0b56f5/60188/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 3430794
 - DQMHistoTests: [Total failures: 65634](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2023-12-05-2300+0b56f5/60188/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3365138
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 214 log files, 167 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_0_X_2023-12-05-2300+0b56f5/60188/triggerResults): found differences in 1 / 48 workflows

jblomer committed:
[ntuple] replace R__ASSERT by assert() in RDF DS

jblomer committed:
[ntuple] Add comments in RDF DS (NFC)

jblomer committed:
[ntuple] change RNTupleColumnReader::fEntryOffset from ULong64_t to Long64_t

jblomer committed:
[ntuple] add RDF factory for datasource with chain

Axel-Naumann committed:
Revert "[Cling] Simplify std::tuple/pair value printer"

This reverts commit bfbb58e6f74c38b883ce6e6e096b6c6e44028869.
cling needs to be able to interpret C++11, even if built with C++17.

vgvassilev committed:
[cling] Add docs how to build from the mono repo

vgvassilev committed:
Prepare for releasing cling v1.0

vgvassilev committed:
Bump cling version to 1.1~dev.

cmsbuild commented:
Pull request #189 was updated.
